tests/functional/snowflake_view_dependency/test_snowflake_view_dependency.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+
+_MODELS__DEPENDENT_MODEL_SQL = """
+
+{% if var('dependent_type', 'view') == 'view' %}
+    {{ config(materialized='view') }}
+{% else %}
+    {{ config(materialized='table') }}
+{% endif %}
+
+select * from {{ ref('base_table') }}
+
+"""
+
+
+
+_MODELS__BASE_TABLE_SQL = """
+
+{{ config(materialized='table') }}
+
+select *
+    {% if var('add_table_field', False) %}
+        , 1 as new_field
+    {% endif %}
+
+from {{ ref('people') }}
+
+"""
+
+_SEEDS__PEOPLE_CSV = """id,name
+1,Drew
+2,Jake
+3,Connor
+"""
+
+class TestSnowflakeLateBindingViewDependency:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dependent_model.sql":  _MODELS__DEPENDENT_MODEL_SQL,
+            "base_table.sql": _MODELS__BASE_TABLE_SQL
+        }
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"people.csv": _SEEDS__PEOPLE_CSV}
+
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            'config-version': 2,
+            'seed-paths': ['seeds'],
+            'seeds': {
+                'quote_columns': False,
+            },
+            'quoting': {
+                'schema': False,
+                'identifier': False
+            }
+        }
+
+    def compare_relations(self, project):
```

Comment:
If this is renamed `def setup_method(self, method, project):`, I believe it runs for each unit test in this class, hence does not need to be called explicitly in the unit test. I'm not sure how it works with the `project` fixture though. If it doesn't work, I would move lines 70-71 into a separate unit test and omit `project` (and `method` as it's optional) as parameters. If it passes once, I don't see how it would fail otherwise. It's not dependent on the contents of the existing unit tests. For more information on setup/teardown methods in pytest, read this: https://docs.pytest.org/en/7.2.x/how-to/xunit_setup.html.

Reply:
Worst case make it a fixture, but the `setup*` methods that come with fixtures are great options